content/teknologi/altinnstudio/altinn-api/app-api/data-element/_index.md

```diff
@@ -0,0 +1,147 @@
+
+**basePath**
+```http
+{org}/{appname}/instances/{instanceOwnerPartyId}/{instanceGuid}/data
+```
+
+## Get data 
+
+Endpoint for downloading the data blob.
+
+```http
+GET basePath/{dataGuid}
+Accept: application/{xml/json}
+```
+
+The response will depend on the type of blob. There are currently 2 primary types; forms and attachments.
+
+### Response with form data
+
+The respons will either be a json or xml serialized version of the object model depending on the **Accept** header in the request. The value **application/xml** will result in an XML document and the value **application/json** will result inn a JSON document.
+
+### Response with instance attachment
+
+The respons will be a file stream. Content-Type will be the same as the original value given when the file was uploaded. The same is true for the file name.
+
+```http
+Content-Disposition: attachment; filename=cute_cat.png; filename*=UTF-8''cute_cat.png
+Content-Length: 16994
+Content-Type: image/png
+```
+
+## Upload data
+
+Endpoint for initially uploading a new data element on a specific instance.
+
+```http
+POST basePath?dataType={data type name}
+```
+
+The **dataType** parameter is required and should contain the name of one of the data types defined on the application. Data types with an **appLogic** property are linked to a form and will have data validation and calculation rules associated with them. Submitted data will be deserialized from **application/xml** or **application/json** based on the content-type of the request. Data types without an **appLogic** property will be handeled as any random file and streamed directly to storage.
+
+### Submitting form as application/json
+
+```json
+Content-Type: application/json
+
+{
+    "dataFormatProvider": "SERES",
+    "dataFormatId": "5703",
+    "dataFormatVersion": "34553",
+    "Tjenesteeier": null,
+    "Kontaktperson": {
+        "navn": "Sophie Salt",
+        "epost": "1337@altinnstudiotestusers.com",
+        "telefonnummer": "90001337"
+    },
+    "OEnsketBruk": null
+}
+
+```
+
+### Submitting form as application/xml
+
+```xml
+Content-Type: application/xml
+
+<BliTjenesteeier_M xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" dataFormatProvider="SERES" dataFormatId="5703" dataFormatVersion="34553">
+    <Kontaktperson>
+        <navn>Sophie Salt</navn>
+        <epost>1337@altinnstudiotestusers.com</epost>
+        <telefonnummer>90001337</telefonnummer>
+    </Kontaktperson>
+</BliTjenesteeier_M>
+```
+
+### Submitting attachment
+
+Most web api client frameworks will have support for creating a valid file upload request for binary data.
+
+```http
+Content-Type: application/pdf
+Content-Disposition: attachment; filename=receipt.pdf; filename*=UTF-8''receipt.pdf
+Content-Length: 16994
+
+%PDF-1.4
+%Óëéá
+1 0 obj
+...
+```
+
+### Upload response
```

Comment:
Going with upload in the header for the previous subsections might require re-thinking this header as they don't really do together.. Will be a bit confusing.